src/gui/sdlmain.cpp

```diff
@@ -3033,59 +3019,63 @@ static SDL_Point clamp_to_minimum_window_dimensions(SDL_Point size)
 }
 
 // Takes in:
-//  - The 'max_resolution' config value: 'auto', 'WxH', or an invalid setting.
+//  - The 'viewport_resolution' config value: 'auto', 'WxH', or an invalid setting.
 //
 // Except for SURFACE and TEXTURE rendering, the function populates the following struct members:
-//  - 'sdl.desktop.use_max_resolution', true if the max_resolution feature is enabled.
-//  - 'sdl.desktop.max_resolution', with the refined size.
+//  - 'sdl.desktop.use_viewport_limits', true if the viewport_resolution feature is enabled.
+//  - 'sdl.desktop.viewport_resolution', with the refined size.
 
-static void setup_max_resolution_from_conf(const std::string &max_resolution_val)
+static void setup_viewport_resolution_from_conf(const std::string &viewport_resolution_val)
 {
-	sdl.use_max_resolution = false;
-	sdl.max_resolution = {-1, -1};
+	sdl.use_viewport_limits = false;
+	sdl.viewport_resolution = {-1, -1};
 
 	// TODO: Deprecate SURFACE output and remove this.
 	if (sdl.desktop.want_type == SCREEN_SURFACE)
 		return;
 
-	if (max_resolution_val == "auto")
+	constexpr auto default_val = "fit";
+	if (viewport_resolution_val == default_val)
 		return;
 
-	const auto desktop = get_desktop_resolution();
-	int w, h;
-	auto was_parsed = sscanf(max_resolution_val.c_str(), "%dx%d", &w, &h) == 2;
+	int w = 0;
+	int h = 0;
+	int p = 0;
+	const auto was_parsed = sscanf(viewport_resolution_val.c_str(), "%dx%d", &w, &h) == 2 ||
+	                        sscanf(viewport_resolution_val.c_str(), "%d%%", &p) == 1;
```

Comment:
Could we allow fractional percentages? (e.g. `20.75%`). It's easy enough to do, so why restrict to integers.

Reply:
Good catch; added!